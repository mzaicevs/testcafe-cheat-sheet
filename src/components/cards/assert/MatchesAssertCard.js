import React from "react";
import { assert_match } from "../../../snippets";
import CodeCard from "../../../base_components/CodeCard";

const MatchesAssertCard = () => (
  <CodeCard
    cardTitle={assert_match.title}
    fullDocLink={assert_match.fullDocLink}
    codeSnippet={{
      code: assert_match.code,
      language: assert_match.language,
      options: assert_match.options,
      example: assert_match.example
    }}
    description={assert_match.description}
  />
);

export default MatchesAssertCard;
