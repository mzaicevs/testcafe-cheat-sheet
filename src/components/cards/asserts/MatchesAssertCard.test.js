import React from "react";
import { shallow } from "enzyme";
import MatchesAssertCard from "./MatchesAssertCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("MatchesAssertCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<MatchesAssertCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
