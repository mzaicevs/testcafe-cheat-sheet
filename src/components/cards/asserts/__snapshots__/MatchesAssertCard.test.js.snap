// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchesAssertCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Match - Assert"
  codeSnippet={
    Object {
      "code": Array [
        "t.expect( actual ).match( re, message, options );",
      ],
      "example": "import { ClientFunction } from 'testcafe';

  fixture \`My fixture\`;
  
  test('My test', async t => {
      const getLocation = ClientFunction(() => document.location.href.toString());
      
      await t
        .expect(getLocation()).match(/.com/)
        .expect('foobar').match(/^f/, 'this assertion will be passed');
  });",
      "language": "javascript",
      "options": "{ 
    timeout: Number,
      allowUnawaitedPromise: Boolean 
  }",
    }
  }
  description="Asserts that actual matches the 're' regular expression."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/assertions/assertion-api.html#match"
/>
`;
